src/deepsparse/pipeline.py

```diff
@@ -119,12 +124,22 @@ def __init__(
         scheduler: Scheduler = None,
         input_shapes: List[List[int]] = None,
         alias: Optional[str] = None,
+        context: Optional[Context] = None,
     ):
         self._model_path_orig = model_path
         self._model_path = model_path
         self._engine_type = engine_type
         self._alias = alias
 
+        self.context = context
+        if self.context is not None:
+            num_cores = num_cores or self.context.num_cores
+            if self.context.num_cores != num_cores:
+                raise ValueError(
+                    f"num_cores mismatch. Expected {self.context.num_cores}, "
```

Comment:
should specify mismatch is between context and Pipeline. would be hard for a user to figure out where the expected val is coming from